pkg/feature/serverless/loaders.go

```diff
@@ -0,0 +1,24 @@
+package serverless
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+)
+
+func ServingIngressDomain(f *feature.Feature) error {
+	domain := strings.TrimSpace(f.Spec.Serving.IngressGateway.Domain)
+	if len(domain) == 0 {
+		var errDomain error
+		domain, errDomain = GetDomain(f.DynamicClient)
+		if errDomain != nil {
+			return fmt.Errorf("failed to fetch OpenShift domain to generate certificate for Serverless: %w", errDomain)
+		}
+
+		domain = "*." + domain
```

Comment:
just out of curiosity, why a wildcard is needed here?

Reply:
If you do `oc get ingresses.config.openshift.io cluster -o yaml` on a cluster, you will notice that `spec.domain` does not have a wildcard. This is the resource where we fetch the domain.

The certificate needs to be valid for subdomains of that `spec.domain`. Thus, we add the wildcard subdomain.